### PR TITLE
docs(vol-restore): added a doc for volume restore from a snapshot

### PR DESCRIPTION
**Pull request by AJDatacore**

### CODE REVIEW

File: reference/snapshot-restore.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "1"
+  thin: "true"
+provisioner: io.openebs.csi-mayastor
+```
+{% endtab %}
+{% endtabs %}
+
+> Note the name of the StorageClass, which, in this example, is **mayastor-1-restore**.
```
Comment (Abhinandan-Purkait) [inline reference/snapshot-restore.md:50]: Please add `Note: The thin: "true" and repl: "1" are the only supported combination.` somewhere here or above this line.

**Review by Abhinandan-Purkait — APPROVED** [review]
Do you need to rebase the branch?

File: reference/snapshot-restore.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+---
+Title: Volume Restore from a Snapshot
+---
+
+Volume restore from an existing snapshot will create exact replicas of storage volumes captured at a specific point in time. They serve as an essential tool for data protection, recovery, and efficient management in Kubernetes environments. This article provides a step-by-step guide on how to create a volume restore.
```
Comment (avishnu) [inline reference/snapshot-restore.md]: Do we need to mention that the replica is a write-able copy?

Comment (avishnu) [reply reference/snapshot-restore.md]: As discussed, the above review need not be addressed.

Comment (avishnu) [reply reference/snapshot-restore.md]: Make it singular as in "exact replica of a storage volume"

**Comment by Abhinandan-Purkait** [comment]
https://mayastor.gitbook.io/introduction/quickstart/configure-mayastor/storage-class-parameters

We need to add `cloneFsIdAsVolumeId` in the above section. The possible values can be `true` and `false`. If not specified its taken as `false`. 

If this is enabled then, the created clone/restore's filesystem uuid would be set to the volume uuid. This feature is needed because filesystem like xfs doesn't allow duplicate filesystem uuid on the same machine by default. 

If this is disabled then we would use the same uuid as that of the parent volume but, the mount would happen using `nouuid` flag which allows duplicate uuid.

The above is a gist of the description. @avishnu Can you please review this content and enhance/modify accordingly?

**Comment by avishnu** [comment]
> https://mayastor.gitbook.io/introduction/quickstart/configure-mayastor/storage-class-parameters
> 
> We need to add `cloneFsIdAsVolumeId` in the above section. The possible values can be `true` and `false`. If not specified its taken as `false`.
> 
> If this is enabled then, the created clone/restore's filesystem uuid would be set to the volume uuid. This feature is needed because filesystem like xfs doesn't allow duplicate filesystem uuid on the same machine by default.
> 
> If this is disabled then we would use the same uuid as that of the parent volume but, the mount would happen using `nouuid` flag which allows duplicate uuid.
> 
> The above is a gist of the description. @avishnu Can you please review this content and enhance/modify accordingly?

Content looks good. Minor change:
- filesystems like XFS do not allow...
- If this is disabled, the uuid of the original volume will be used for the restore filesystem as well, but the restore filesystem will be mounted using the `nouuid` flag to bypass the duplicate uuid validation.

File: reference/storage-class-parameters.md [hunk 3]
```diff
@@ -75,3 +75,10 @@ The `stsAffinityGroup` ensures that in such cases, the targets are distributed o
 
 By default, the `stsAffinityGroup` feature is disabled. To enable it, modify the storage class YAML by setting the `parameters.stsAffinityGroup` parameter to true.
 
+## "cloneFsIdAsVolumeId"
+
+`cloneFsIdAsVolumeId` is a setting for volume clones with two options: `true` and `false`. By default, it is set to `false`.
+- When set to `true`, the created clone/restore's filesystem `uuid` will be set to the new volume `uuid`. This is important because some file systems, like XFS, don't allow duplicate filesystem `uuid` on the same machine by default.
```
Comment (Abhinandan-Purkait) [inline reference/storage-class-parameters.md]: new volume `uuid`

Is new volume confusing? What would be better here? Basically it should be equal to the created clone/restore's id.

**Review by Abhinandan-Purkait — APPROVED** [review]

File: reference/storage-class-parameters.md [hunk 4]
```diff
@@ -49,7 +49,7 @@ The `agents.core.capacity.thin` spec present in the Mayastor helm chart consists
 {% hint style="info" %}
 Note:
 1. By default, the volumes are provisioned as `thick`. 
-2. For a pool of a particular size, say 10 Gigabytes, a volume > 10 Gigabytes cannot be created, as Mayastor 2.3.0 does not support pool expansion.
+2. For a pool of a particular size, say 10 Gigabytes, a volume > 10 Gigabytes cannot be created, as Mayastor 2.4.0 does not support pool expansion.
```
Comment (avishnu) [inline reference/storage-class-parameters.md]: We can remove version from here.
Just say, "as Mayastor currently does not support pool expansion"

File: reference/storage-class-parameters.md (hunk 3 above)
Comment (avishnu) [reply reference/storage-class-parameters.md]: Ok, I can see 2 changes:
- setting for volume clones/restores
- will be set to the restore volume's uuid

File: reference/storage-class-parameters.md [hunk 5]
```diff
@@ -77,8 +77,8 @@ By default, the `stsAffinityGroup` feature is disabled. To enable it, modify the
 
 ## "cloneFsIdAsVolumeId"
 
-`cloneFsIdAsVolumeId` is a setting for volume clones with two options: `true` and `false`. By default, it is set to `false`.
-- When set to `true`, the created clone/restore's filesystem `uuid` will be set to the new volume `uuid`. This is important because some file systems, like XFS, don't allow duplicate filesystem `uuid` on the same machine by default.
+`cloneFsIdAsVolumeId` is a setting for volume clones/restores with two options: `true` and `false`. By default, it is set to `false`.
+- When set to `true`, the created clone/restore's filesystem `uuid` will be set to the restore volume's `uuid`. This is important because some file systems, like XFS, don't allow duplicate filesystem `uuid` on the same machine by default.
```
Comment (avishnu) [inline reference/storage-class-parameters.md]: "do not" in place of "don't" ?

**Review by avishnu — APPROVED** [review]
Lgtm

File: SUMMARY.md [hunk 6]
```diff
@@ -23,6 +23,7 @@
 * [Node Cordon](reference/node-cordon.md)
 * [Node Drain](reference/node-drain.md)
 * [Volume Snapshots](reference/snapshot.md)
+* [Volume Restore from Snapshot](reference\snapshot.md)
```
Comment (avishnu) [inline SUMMARY.md:26]: Shouldn't this be snapshot-restore.md?

**Pull request merged**